substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java

```diff
@@ -46,24 +60,26 @@ public static int utf8Length(String string) {
     /**
      * @param beginIndex first index that is part of the region, inclusive
      * @param endIndex index at the end of the region, exclusive
-     * @return the length in bytes of the UTF8 representation of the string region
+     * @return the length as {@code int} in bytes of the UTF8 representation of the string region.
+     *         Might return a truncated size if the value does not fit into {@code int} (see
+     *         JDK-8328877).
      */
     public static int utf8Length(String s, int beginIndex, int endIndex) {
+        // Based on
+        // https://github.com/openjdk/jdk21u/blob/jdk-21.0.4%2B7/src/hotspot/share/utilities/utf8.cpp#L433-L444
```

Comment:
Should use the similar style than others hunks in svm.core:

```suggestion
        // @BasedOnJDKFile("https://github.com/openjdk/jdk21u/blob/jdk-21.0.4%2B7/src/hotspot/share/utilities/utf8.cpp#L433-L444")
        public static int utf8Length(String s, int beginIndex, int endIndex) {
```

Reply:
https://github.com/oracle/graal/commit/55609a7014b92a635d29e78b2a252a7417440301#diff-b37fd4151e56ac177ddece5693f781b50b9b6f333d86583bad58704764364fcc <- `objectfile` doesn't have annotation...

Reply:
Fair enough. This will be hard to remember.